tests/pyfunc/test_scoring_server.py

```diff
@@ -429,6 +444,14 @@ def test_split_oriented_json_to_df():
     assert set(str(dt) for dt in df.dtypes) == {"object", "float64", "int64"}
 
 
+def test_parse_with_schema_csv(pandas_df_with_csv_types):
+    schema = Schema([ColSpec(c, c) for c in pandas_df_with_csv_types.columns])
+    df = _shuffle_pdf(pandas_df_with_csv_types)
+    csv_str = df.to_csv(index=False)
+    df = pyfunc_scoring_server.parse_csv_input(csv_str, schema=schema)
```

Comment:
We need StringIO here:
```suggestion
    df = pyfunc_scoring_server.parse_csv_input(csv_str, schema=schema)
```

Please import it using 
```
try:
    from StringIO import StringIO
except ImportError:
    from io import StringIO
```
to maintain compatibility.